back/app/models/user.rb

```diff
@@ -380,13 +382,11 @@ def active?
   end
 
   def blocked?
-    block_start_at.present? && block_start_at.between?(block_duration.days.ago, Time.zone.now)
+    block_end_at.present? && block_end_at > Time.zone.now
   end
 
-  def block_end_at
-    return nil unless blocked?
-
-    block_start_at + block_duration.days
+  def active_or_blocked?
```

Comment:
Would `registered?` or `completed_registration?` work here?

Reply:
And then

```ruby
  def active?
    registered? && !blocked?
  end
```

Reply:
Maybe. `active?` includes`!invite_pending?`, but I don't think you can have `registration_completed_at.present?` if you have `invite_pending?` anyway 🤔

Reply:
But I might include it anyway, to avoid some weird edge case we havn't though of. So we could have...
```
def registered?
  registration_completed_at.present? && !invite_pending?
end

Reply:
I thought I could get away with removing `!invite_pending` from the picture, but it seems it would require work that may not be useful in the light of planned changes by the methods squad. (See [Slack thread](https://citizenlabco.slack.com/archives/C016C2EHURY/p1678992484306909
))

Reply:
I think it's the right decision to leave `!invite_pending`. Definitely, it was added for a reason.

Reply:
Changed approach and removed it. See [comment](https://github.com/CitizenLabDotCo/citizenlab/pull/4252#issuecomment-1474102081).